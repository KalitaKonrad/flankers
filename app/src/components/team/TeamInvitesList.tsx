import * as React from 'react';
import { Dispatch, MutableRefObject, SetStateAction } from 'react';
import {
  FlatList,
  ListRenderItemInfo,
  StyleProp,
  StyleSheet,
  View,
  ViewStyle,
} from 'react-native';
import { List } from 'react-native-paper';
import BottomSheet from 'reanimated-bottom-sheet';

import { InvitePayload } from '../../types/inviteResponse';
import { AppButton } from '../shared/AppButton';
import { Avatar } from '../shared/Avatar';

interface TeamInvitesProps {
  invites: InvitePayload[];
  modalRef: MutableRefObject<BottomSheet | null>;
  setAcceptToken: Dispatch<SetStateAction<string>>;
  setDeclineToken: Dispatch<SetStateAction<string>>;
  style?: StyleProp<ViewStyle>;
}

export const TeamInvitesList: React.FC<TeamInvitesProps> = ({
  invites,
  style,
  modalRef,
  setAcceptToken,
  setDeclineToken,
}) => {
  const onInviteClick = (acceptToken: string, declineToken: string) => {
    modalRef?.current?.snapTo(0);
    setAcceptToken(acceptToken);
    setDeclineToken(declineToken);
  };

  const renderItem = ({ item }: ListRenderItemInfo<InvitePayload>) => (
    <List.Item
      accessibilityTraits=""
      accessibilityComponentType
      title={item.team_name}
      titleStyle={styles.invitedByEmail}
      description={item.team_description}
      style={styles.itemContainer}
      left={() => (
        <View style={styles.memberAvatarContainer}>
          <Avatar size={40} borderRadius={8} src={{ uri: item.team_avatar }} />
        </View>
      )}
      right={() => (
        <View style={styles.buttonContainer}>
          <AppButton
            onPress={() => onInviteClick(item.accept_token, item.deny_token)}>
            Dołącz
          </AppButton>
        </View>
      )}
    />
  );

  return (
    <FlatList
      data={invites}
      contentContainerStyle={styles.container}
      renderItem={renderItem}
      keyExtractor={(item) => item.id.toString()}
    />
  );
};

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 16,
  },
  itemContainer: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    marginVertical: 8,
    height: 80,
  },
  buttonContainer: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  invitedByEmail: {
    fontWeight: 'bold',
  },
  memberAvatarContainer: {
    justifyContent: 'center',
    marginRight: 4,
  },
});
