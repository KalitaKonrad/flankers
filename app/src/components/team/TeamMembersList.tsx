import React from 'react';
import {
  FlatList,
  ListRenderItemInfo,
  StyleProp,
  StyleSheet,
  View,
  ViewStyle,
} from 'react-native';
import { List } from 'react-native-paper';
import SkeletonPlaceholder from 'react-native-skeleton-placeholder';

import { UserProfilePayload } from '../../types/userProfilePayload';
import { Avatar } from '../shared/Avatar';

interface TeamMemberListProps {
  members: UserProfilePayload[];
  style?: StyleProp<ViewStyle>;
  isLoading: boolean;
}

export const TeamMemberList: React.FC<TeamMemberListProps> = ({
  members,
  style,
  isLoading,
}) => {
  const renderItem = ({ item }: ListRenderItemInfo<UserProfilePayload>) =>
    isLoading ? (
      <SkeletonPlaceholder>
        <View style={styles.container} />
      </SkeletonPlaceholder>
    ) : (
      <List.Item
        title={item.name}
        titleStyle={styles.memberName}
        description={`${item.elo} punktÃ³w rankingowych`}
        left={() => (
          <View style={styles.memberAvatarContainer}>
            <Avatar
              size={40}
              borderRadius={8}
              src={{ uri: item.versioned_avatar }}
              isLoading={isLoading}
            />
          </View>
        )}
      />
    );

  return (
    <FlatList
      data={members}
      contentContainerStyle={styles.container}
      renderItem={renderItem}
      keyExtractor={(member) => member.id.toString()}
    />
  );
};

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 16,
  },
  memberName: {
    fontWeight: 'bold',
  },

  memberAvatarContainer: {
    justifyContent: 'center',
    marginRight: 4,
  },
});
