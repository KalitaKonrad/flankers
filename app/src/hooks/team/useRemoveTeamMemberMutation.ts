import { useMutation, useQueryClient } from 'react-query';

import {
  QUERY_PROFILE_KEY,
  QUERY_TEAM_KEY,
  QUERY_TEAM_MEMBERS,
} from '../../const/query.const';
import { useAxios } from '../useAxios';

interface RemoveTeamMemberPayload {
  team_id: string;
  user_id: number;
}

export const useRemoveTeamMemberMutation = () => {
  const axios = useAxios();

  const queryCache = useQueryClient();

  return useMutation(
    ({ team_id, user_id }: RemoveTeamMemberPayload) => {
      return axios.delete(`teams/memberships/${team_id}`, {
        data: { user_id },
      });
    },
    {
      onSuccess: async () => {
        const members = queryCache.invalidateQueries(QUERY_TEAM_MEMBERS);
        const team = queryCache.invalidateQueries(QUERY_TEAM_KEY);
        const profile = queryCache.invalidateQueries(QUERY_PROFILE_KEY);
        const refetch = queryCache.refetchQueries(
          [QUERY_TEAM_MEMBERS, QUERY_PROFILE_KEY, QUERY_TEAM_KEY],
          { exact: true }
        );

        await Promise.all([members, team, profile]);
        await refetch;
      },
    }
  );
};
