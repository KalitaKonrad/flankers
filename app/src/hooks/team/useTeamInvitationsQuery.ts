import { useQuery } from 'react-query';

import { QUERY_TEAM_INVITATIONS_ALL } from '../../const/query.const';
import { InvitePayload } from '../../types/inviteResponse';
import { useAxios } from '../useAxios';

export type InviteResponse = { data: InvitePayload[] };

export const useTeamInvitationsQuery = () => {
  const axios = useAxios();

  return useQuery<InviteResponse>(QUERY_TEAM_INVITATIONS_ALL, async () => {
    const response = await axios.get('teams/invites');
    return response.data;
  });
};
