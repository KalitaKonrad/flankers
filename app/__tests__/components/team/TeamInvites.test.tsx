import { render } from '@testing-library/react-native';
import * as React from 'react';

import { TeamInvitesList } from '../../../src/components/team/TeamInvitesList';

describe('<TeamInvitesList>', () => {
  it('renders correctly when empty array with data provided', () => {
    const tree = render(
      <TeamInvitesList
        invites={[]}
        modalRef={null as any}
        setAcceptToken={() => null}
        setDeclineToken={() => null}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when all data provided', () => {
    const mockData = [
      {
        id: 5,
        user_id: 5,
        team_id: 5,
        type: 'string',
        email: 'string',
        accept_token: 'string',
        deny_token: 'string',
        created_at: 'string',
        updated_at: 'string',
        team_name: 'string',
        team_avatar: 'string',
        team_description: 'string',
      },
    ];

    const tree = render(
      <TeamInvitesList
        invites={mockData}
        modalRef={null as any}
        setAcceptToken={() => null}
        setDeclineToken={() => null}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
