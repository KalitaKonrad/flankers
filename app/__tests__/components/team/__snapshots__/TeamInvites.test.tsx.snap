// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeamInvitesList> renders correctly when all data provided 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "paddingHorizontal": 16,
    }
  }
  data={
    Array [
      Object {
        "accept_token": "string",
        "created_at": "string",
        "deny_token": "string",
        "email": "string",
        "id": 5,
        "team_avatar": "string",
        "team_description": "string",
        "team_id": 5,
        "team_name": "string",
        "type": "string",
        "updated_at": "string",
        "user_id": 5,
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            false,
            Array [
              Object {
                "padding": 8,
              },
              Object {
                "alignItems": "center",
                "display": "flex",
                "height": 80,
                "justifyContent": "center",
                "marginVertical": 8,
              },
            ],
          ]
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "justifyContent": "center",
                "marginRight": 4,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#fff",
                  },
                  Object {
                    "borderRadius": 8,
                    "elevation": 0,
                    "height": 40,
                    "width": 40,
                  },
                  Object {
                    "shadowColor": "rgba(0, 0, 0, 0.1)",
                    "shadowOffset": Object {
                      "height": 4,
                      "width": 0,
                    },
                    "shadowOpacity": 1,
                    "shadowRadius": 4,
                  },
                  undefined,
                ]
              }
            >
              <Image
                source={
                  Object {
                    "uri": "string",
                  }
                }
                style={
                  Array [
                    Object {
                      "borderColor": "#FFFFFF",
                    },
                    Object {
                      "borderRadius": 8,
                      "borderWidth": 0,
                      "height": 40,
                      "width": 40,
                    },
                  ]
                }
              />
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "marginVertical": 6,
                  "paddingLeft": 8,
                },
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "fontSize": 16,
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                    },
                    Object {
                      "fontWeight": "bold",
                    },
                  ],
                ]
              }
            >
              string
            </Text>
            <Text
              numberOfLines={2}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "fontSize": 14,
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.54)",
                    },
                    undefined,
                  ],
                ]
              }
            >
              string
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "justifyContent": "center",
              }
            }
          >
            <View
              accessibilityComponentType=""
              accessibilityTraits=""
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderColor": "transparent",
                  "borderRadius": 100,
                  "borderStyle": "solid",
                  "borderWidth": 0,
                  "elevation": 0,
                  "minWidth": 64,
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "disabled": undefined,
                  }
                }
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 100,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      Array [
                        Object {
                          "height": 56,
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#000000",
                          "fontFamily": "System",
                          "fontWeight": "400",
                          "textAlign": "left",
                        },
                        Array [
                          Object {
                            "letterSpacing": 1,
                            "marginHorizontal": 16,
                            "marginVertical": 9,
                            "textAlign": "center",
                          },
                          undefined,
                          false,
                          Object {
                            "color": "#6200ee",
                            "fontFamily": "System",
                            "fontWeight": "500",
                          },
                          Object {
                            "fontFamily": "System",
                            "fontWeight": "500",
                          },
                          Array [
                            Object {
                              "fontSize": 16,
                              "fontWeight": "bold",
                              "letterSpacing": 0,
                            },
                            false,
                            undefined,
                          ],
                        ],
                      ]
                    }
                  >
                    Dołącz
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`<TeamInvitesList> renders correctly when empty array with data provided 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "paddingHorizontal": 16,
    }
  }
  data={Array []}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View />
</RCTScrollView>
`;
